core/dbt/task/init.py

```diff
@@ -40,40 +44,68 @@
 Happy modeling!
 """
 
+# https://click.palletsprojects.com/en/8.0.x/api/?highlight=float#types
+click_type_mapping = {
+    "string": click.STRING,
+    "int": click.INT,
+    "float": click.FLOAT,
+    "bool": click.BOOL,
+    None: None
+}
+
 
 class InitTask(BaseTask):
     def copy_starter_repo(self, project_name):
         logger.debug("Starter project path: " + starter_project_directory)
         shutil.copytree(starter_project_directory, project_name,
                         ignore=shutil.ignore_patterns(*IGNORE_FILES))
 
-    def create_profiles_dir(self, profiles_dir):
-        if not os.path.exists(profiles_dir):
+    def create_profiles_dir(self, profiles_dir: str) -> bool:
+        """Create the user's profiles directory if it doesn't already exist."""
+        profiles_path = Path(profiles_dir)
+        if profiles_path.exists():
             msg = "Creating dbt configuration folder at {}"
             logger.info(msg.format(profiles_dir))
             dbt.clients.system.make_directory(profiles_dir)
             return True
         return False
 
-    def create_profiles_file(self, profiles_file, sample_adapter):
+    def create_profile_from_sample(self, adapter: str, profile_name: str):
+        """Create a profile entry using the adapter's sample_profiles.yml
+
+        Renames the profile in sample_profiles.yml to match that of the project."""
         # Line below raises an exception if the specified adapter is not found
-        load_plugin(sample_adapter)
-        adapter_path = get_include_paths(sample_adapter)[0]
-        sample_profiles_path = adapter_path / 'sample_profiles.yml'
+        load_plugin(adapter)
+        adapter_path = get_include_paths(adapter)[0]
+        sample_profiles_path = adapter_path / "sample_profiles.yml"
 
         if not sample_profiles_path.exists():
-            logger.debug(f"No sample profile found for {sample_adapter}, skipping")
-            return False
-
-        if not os.path.exists(profiles_file):
-            msg = "With sample profiles.yml for {}"
-            logger.info(msg.format(sample_adapter))
-            shutil.copyfile(sample_profiles_path, profiles_file)
-            return True
-
-        return False
+            logger.debug(f"No sample profile found for {adapter}.")
+        else:
+            with open(sample_profiles_path, "r") as f:
```

Comment:
Here and elsewhere, we probably want to lead on the `dbt.clients.system` module for cross-OS support / error handling. I'm thinking about `load_file_contents` and `write_file` in particular... though we'll need to adjust that method to support the `r+` and `a` modes you're using here.

cc @leahwicz: I'm pretty fuzzy on this stuff, so you should correct me if I'm wrong here. My sense is, the more we can lean on `clients.system` methods for all file operations, the better-served we'll be in a world with storage adapters etc. Alternatively, we could simply say that since `init` is a CLI-only task, we don't care about needing to support it outside of the local file system.